### PR TITLE
docs(dart): convert toh-5 to Dart

### PR DESCRIPTION
**Pull request by kwalrath**
Changes based on @kasperpeulen's #700.

I also found some `"`s in various toh-*/*.dart files, which I converted to `'`s.

@chalin could you please take a look? I seriously considered converting to your new way of including the TS file, but in the end decided to just get it out and put up what I already had.

### CODE REVIEW

File: public/docs/dart/latest/tutorial/toh-pt5.jade [hunk 1]
```diff
@@ -1,14 +1,696 @@
...
+
+    Now implement the `DashboardComponent` class like this:
++makeExample('toh-5/dart/lib/dashboard_component_2.dart','component', 'lib/dashboard_component.dart (class)')
+:marked
+    We saw this kind of logic before in the `HeroesComponent`.
+    * create a `heroes` list property
+    * inject the `HeroService` in the constructor and hold it in a private `_heroService` field.
+    * call the service to get heroes inside the Angular `ngOnInit` lifecycle hook.
+
+    The noteworthy differences: we cherry-pick four heroes (2nd, 3rd, 4th, and 5th) with *slice*
```
Comment (chalin) [inline public/docs/dart/latest/tutorial/toh-pt5.jade]: The "with *slice*" suggests there is a `slice` operation in Dart. Maybe drop that fragment or just leave it for now and we can address it on the next round (when the doc is refactored to conform to the new design).

File: public/docs/dart/latest/tutorial/toh-pt5.jade [hunk 2]
```diff
@@ -1,14 +1,696 @@
...
+
+.l-main-section
+:marked
+    ## Select a Hero in the *HeroesComponent*
+    We'll do something similar in the `HeroesComponent`.
+
+    That component's current template exhibits a "master/detail" style with the list of heroes
+    at the top and details of the selected hero below.
+
+    **[TODO: Add example, once it exists.]**
```
Comment (chalin) [inline public/docs/dart/latest/tutorial/toh-pt5.jade]: Example needed, which we can address in the next round.

File: public/docs/dart/latest/tutorial/toh-pt5.jade [hunk 3]
```diff
@@ -1,14 +1,696 @@
...
 
-  * [Dart source code](https://github.com/angular/angular.io/tree/master/public/docs/_examples/toh-5/dart):
-    A preliminary Dart version of the Tour of Heroes app,
-    featuring the hero editor, a master/detail page,
-    multiple components, services, and routing
+    ### The Road Ahead
+    We have much of the foundation we need to build an application.
+    We're still missing a key piece: remote data access.
 
+    In a forthcoming tutorial chapter,
```
Comment (chalin) [inline public/docs/dart/latest/tutorial/toh-pt5.jade]: (For the record, toh-6 has landed.)

**Comment by chalin** [comment]
I'd say that this is good to go. We can address the other issues in the next pass.

**Comment by kwalrath** [comment]
Thanks, Patrice. I'll commit this and create an issue so we don't forget to make these fixes.

File: public/docs/dart/latest/tutorial/toh-pt5.jade (hunk 1 above)
Comment (chalin) [inline public/docs/dart/latest/tutorial/toh-pt5.jade]: Fixed

File: public/docs/dart/latest/tutorial/toh-pt5.jade (hunk 2 above)
Comment (chalin) [inline public/docs/dart/latest/tutorial/toh-pt5.jade]: Fixed

File: public/docs/dart/latest/tutorial/toh-pt5.jade (hunk 3 above)
Comment (chalin) [inline public/docs/dart/latest/tutorial/toh-pt5.jade]: Fixed

**Pull request merged**